libraries/AP_Logger/AP_Logger_MAVLink.cpp

```diff
@@ -57,6 +57,10 @@ void AP_Logger_MAVLink::Init()
 
 bool AP_Logger_MAVLink::logging_failed() const
 {
+    // Ensure check doesnt fail for special case: Plane disarmed & logging while disarmed disabled
+    if (!hal.util->get_soft_armed() && _front._params.log_disarmed == AP_Logger::LogDisarmed::NONE) { 
```

Comment:
```suggestion
    if (_front._params.log_disarmed == AP_Logger::LogDisarmed::NONE && !hal.util->get_soft_armed()) { 
```

shortcut-evaluation means that if the comparison fails we don't need to do the hard work of that method call.

It's a minot optimisation, but a good pattern to have.